server/src/main/java/org/apache/cassandra/sidecar/routes/cdc/ListCDCDirHandler.java

```diff
@@ -0,0 +1,145 @@
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.json.Json;
+import io.vertx.core.net.SocketAddress;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.cassandra.sidecar.common.response.ListCDCSegmentResponse;
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.config.ServiceConfiguration;
+import org.apache.cassandra.sidecar.config.SidecarConfiguration;
+import org.apache.cassandra.sidecar.routes.AbstractHandler;
+import org.apache.cassandra.sidecar.utils.CDCUtil;
+import org.apache.cassandra.sidecar.utils.CassandraInputValidator;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+
+import static org.apache.cassandra.sidecar.utils.CDCUtil.getIdxFileName;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.getLogFilePrefix;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.isIndexFile;
+import static org.apache.cassandra.sidecar.utils.CDCUtil.parseIndexFile;
+
+/**
+ * Provides REST endpoint for listing commit logs in CDC directory.
+ */
+@Singleton
+public class ListCDCDirHandler extends AbstractHandler<Void>
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ListCDCDirHandler.class);
+    private final ServiceConfiguration config;
+
+    @Inject
+    public ListCDCDirHandler(InstanceMetadataFetcher metadataFetcher,
+                             SidecarConfiguration config,
+                             ExecutorPools executorPools,
+                             CassandraInputValidator validator)
+    {
+        super(metadataFetcher, executorPools, validator);
+        this.config = config.serviceConfiguration();
+    }
+
+    @Override
+    protected void handleInternal(RoutingContext context,
+                                  HttpServerRequest httpRequest,
+                                  String host,
+                                  SocketAddress remoteAddress,
+                                  Void request)
+    {
+        String cdcDir = metadataFetcher.instance(host).cdcDir();
+        ListCDCSegmentResponse listCDCSegmentResponse = new ListCDCSegmentResponse(config.host(),
+                                                                                   config.port(),
+                                                                                   new LinkedList<>());
+        try
+        {
+            addResources(cdcDir, listCDCSegmentResponse);
+        }
+        catch (IOException e)
+        {
+            LOGGER.warn("Error listing the CDC commit log segments", e);
+            context.response()
+                   .setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code())
+                   .setStatusMessage(e.getMessage())
+                   .end();
+            return;
+        }
+
+        context.response().end(Json.encodePrettily(listCDCSegmentResponse));
+    }
+
+    private void addResources(final String cdcDirPath, final ListCDCSegmentResponse listResponse) throws IOException
```

Comment:
let's not use final here:
```suggestion
    private void addResources(String cdcDirPath, ListCDCSegmentResponse listResponse) throws IOException
```